pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -3819,8 +3842,19 @@ def process_data(self):
         self.proc_data_dict['qscale_data'] = OrderedDict()
         for qbn in self.qb_names:
             self.proc_data_dict['qscale_data'][qbn] = OrderedDict()
-            sweep_points = self.proc_data_dict['sweep_points_dict'][qbn][
-                'msmt_sweep_points']
+            sweep_points = deepcopy(self.proc_data_dict['sweep_points_dict'][
+                                        qbn]['msmt_sweep_points'])
+            unique_sp = np.unique(sweep_points[:3])
```

Comment:
For the qscale measurement, each sweep point needs to be repeated 3 times. So what I do here is I check if the sweep points are already repeated 3 times by looking if the first 3 entries are identical (as they would need to be) or not. The issue is that in the single qubit routine we already save the repeated sweep points in metadata while in the new multi-qubit routine we do not (this is because of the new SweepPoints object).

Reply:
ok, I see. Sorry I didn't see this was in the qscale analysis (the code was folded and I thought it was the baseanalysis). Maybe then a small comment saying that it is for backwards compatibility with single qb routine and than it can be removed once the latter is re-written with the new Sweeppoint objects would be useful. Something like : `# Verifies whether sweep points are stored with repetition or not. For backwards compatibility with QudevTransmon.measure_qscale() that does not (yet) use Sweeppoints object.`

Reply:
I think my main point was to show why we needed to check i.e. the backwards compatibility so that once we change it in measure_qscale we can remove this check. As this was just a really minor thing I committed myself, hope that's ok for you.